net/interfaces/interfaces_bsd.go

```diff
@@ -37,85 +39,118 @@ func defaultRoute() (d DefaultRouteDetails, err error) {
 }
 
 func DefaultRouteInterfaceIndex() (int, error) {
+	rms, err := defaultRouteMessages(1)
+	if err != nil {
+		return 0, err
+	}
+	if len(rms) == 0 {
+		return 0, errors.New("no gateway index found")
+	}
+	return rms[0].Index, nil
+}
+
+func init() {
+	likelyHomeRouterIP = likelyHomeRouterIPBSDFetchRIB
+}
+
+func likelyHomeRouterIPBSDFetchRIB() (ret netip.Addr, ok bool) {
+	rms, err := defaultRouteMessages(math.MaxInt)
```

Comment:
MaxInt is a little overkill. Also for the math package. Also because Go's int size (which varies by platform) might not match the kernel's anyway. I doubt you'd get more than 32 bits of routes from the kernel. Maybe just write 100 or something reasonable here?